### PR TITLE
Fix PageRouteTransformerConvention with custom page routes

### PR DESCRIPTION
**Pull request by JamesNK**
Added a test, found a bug, fixed the bug!

### CODE REVIEW

File: test/WebSites/RoutingWebSite/Pages/PageRouteTransformer/PageWithConfiguredRoute.cshtml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿@page "ExtraPath/{id?}"
+@{
```
Comment (pranavkm) [inline test/WebSites/RoutingWebSite/Pages/PageRouteTransformer/PageWithConfiguredRoute.cshtml]: Remove?

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs [hunk 2]
```diff
@@ -129,8 +129,10 @@ private static string TransformPageRoute(PageRouteModel model, SelectorModel sel
             var pageRouteMetadata = selectorModel.EndpointMetadata.OfType<PageRouteMetadata>().SingleOrDefault();
             if (pageRouteMetadata == null)
             {
-                // Selector does not have expected metadata. Should never reach here
-                throw new InvalidOperationException("Page selector did not have page route metadata.");
+                // Selector does not have expected metadata
+                // This selector was likely configured by AddPageRouteModelConvention
+                // Use the existing explicitly configured template
+                return selectorModel.AttributeRouteModel.Template;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: Should we do this? I'm not sure it makes much sense to transform a template that someone wrote.

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: We shouldn't.

This convention applies the transformer to the folder and file names. The route explicitly configured with the page directive is untouched. Not touching an explicitly defined route in the Startup.cs is consistent with that.

I think it is good to go how it is.

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: We're not. `return selectorModel.AttributeRouteModel.Template;` here is returning the previously configured route. Processing happens below this.

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: Explicitly configured route: https://github.com/aspnet/Mvc/blob/63a51c47cbd537072a75ce7bce8d766a26b5a87d/test/WebSites/RoutingWebSite/Startup.cs#L31
Calling the unchanged explicitly configured route: https://github.com/aspnet/Mvc/blob/63a51c47cbd537072a75ce7bce8d766a26b5a87d/test/Microsoft.AspNetCore.Mvc.FunctionalTests/RoutingTestsBase.cs#L154

The bug was it was previously throwing an exception.

Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: Oh, derp. YEah, so you're pointing out that we *don't* rewrite it. Ok good, then we agreed all along, but I should get better glasses.

**Pull request merged**